lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/evaluator/EvaluatorShim.java

```diff
@@ -0,0 +1,215 @@
+  }
+
+  private void onStop() {
+    LOG.log(Level.FINER, "Entering EvaluatorShim.onStop().");
+
+    if (this.evaluatorProcess != null) {
+      try {
+        LOG.log(Level.INFO, "Waiting for the evaluator process to exit.");
+        this.evaluatorProcess.waitFor();
+
+        LOG.log(Level.INFO, "Destroying the Evaluator Process.");
+        this.evaluatorProcess.destroy();
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    LOG.log(Level.INFO, "Shutting down the evaluatorLaunchExecutorService.");
+    this.evaluatorLaunchExecutorService.shutdown();
+
+    try {
+      LOG.log(Level.INFO, "Closing EvaluatorShim Control channel.");
+      this.evaluatorShimCommandChannel.close();
+    } catch (Exception e) {
+      LOG.log(Level.SEVERE, "An unexpected exception occurred while attempting to close the EvaluatorShim " +
+          "control channel.");
+      throw new RuntimeException(e);
+    }
+
+    try {
+      LOG.log(Level.INFO, "Closing the Remote Manager.");
+      this.remoteManager.close();
+    } catch (Exception e) {
+      LOG.log(Level.SEVERE, "Failed to close the RemoteManager with the following exception: {0}.", e);
+    }
+
+    LOG.log(Level.INFO, "Exiting EvaluatorShim.onStop().");
+  }
+
+  private void onEvaluatorLaunch(final String launchCommand, final String evaluatorConfigString) {
+    LOG.log(Level.FINEST, "Entering EvaluatorShim.onEvaluatorLaunch().");
+
+    File evaluatorConfigurationFile = new File("local/evaluator.conf");
+    LOG.log(Level.FINER, "Persisting evaluator config at: {0}", evaluatorConfigurationFile.getAbsolutePath());
+
+    try {
+      boolean newFileCreated = evaluatorConfigurationFile.createNewFile();
+      LOG.log(Level.FINE,
+          newFileCreated ? "Created a new file for persisting evaluator configuration at {0}."
+              : "Using existing file for persisting evaluator configuration at {0}.",
+          evaluatorConfigurationFile.getAbsolutePath());
+
+      Configuration evaluatorConfiguration = this.configurationSerializer.fromString(evaluatorConfigString);
+      this.configurationSerializer.toFile(evaluatorConfiguration, evaluatorConfigurationFile);
+    } catch (final IOException | BindException e) {
+      LOG.log(Level.SEVERE, "An unexpected exception occurred while attempting to deserialize and write " +
+          "Evaluator configuration file. {0}", e);
+      throw new RuntimeException("Unable to write configuration.", e);
+    }
+
+    LOG.log(Level.INFO, "Launching the evaluator by invoking the following command: " + launchCommand);
+
+    this.evaluatorLaunchExecutorService.submit(new Thread() {
+      public void run() {
+        try {
+          final List<String> command = Arrays.asList(launchCommand.split(" "));
+          EvaluatorShim.this.evaluatorProcess = new ProcessBuilder()
+              .command(command)
+              .redirectError(new File(fileNames.getEvaluatorStderrFileName() + ".txt"))
```

Comment:
Why add the ".txt"

Reply:
Because Azure website will only display files with ".txt". It's kind of a pain needing to download logs before you can view them.

Reply:
Fair enough. Might add a comment stating this.